symfony/src/Service/Api/DataPersister/Auth/PasswordResetDataPersister.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace App\Service\Api\DataPersister\Auth;
+
+use ApiPlatform\Core\DataPersister\DataPersisterInterface;
+use ApiPlatform\Core\Validator\ValidatorInterface;
+use App\ApiResource\Auth\PasswordReset;
+use App\Dto\General\StatusDto;
+use App\Mail\MailHelper;
+use App\Repository\DeviceRepository;
+use App\Repository\UserRepository;
+use App\Service\Website\User\UserService;
+use App\Utils\AuthUtils;
+use Doctrine\ORM\EntityManagerInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Class PasswordResetDataPersister
+ *
+ * This is a custom DataPersister for which incoming data can be handled, persisted and customized in any way.
+ * More information: https://api-platform.com/docs/core/data-persisters.
+ *
+ * @package App\Service\Api\DataPersister\Auth
+ */
+final class PasswordResetDataPersister implements DataPersisterInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function __construct(
+        private EntityManagerInterface $entityManager,
+        private UserPasswordEncoderInterface $userPasswordEncoder,
+        private ValidatorInterface $validator,
+        private DeviceRepository $deviceRepository,
+        private Security $security,
+        private UserService $userService,
+        private UserRepository $userRepository,
+        private MailHelper $mailHelper,
+        private LoggerInterface $logger,
+        private TranslatorInterface $translator
+    ) {
+        //
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function supports($data): bool
+    {
+        return $data instanceof PasswordReset;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function persist($data)
+    {
+        // Validate input.
+        $context["groups"] = "password-reset";
+        $this->validator->validate($data, $context);
+
+        // Default response.
+        $output = new StatusDto();
+        $output->code = Response::HTTP_CREATED;
+        $output->message = 'Password reset mail sent to user.';
```

Comment:
Best ineens voor alle responses translations gebruiken => eenvoudiger bij updaten of multilingual support